src/detect.c

```diff
@@ -766,9 +767,11 @@ static inline void DetectRulePacketRules(
 
         if (unlikely(sflags & SIG_FLAG_DSIZE)) {
             if (likely(p->payload_len < s->dsize_low || p->payload_len > s->dsize_high)) {
-                SCLogDebug("kicked out as p->payload_len %u, dsize low %u, hi %u",
-                        p->payload_len, s->dsize_low, s->dsize_high);
-                goto next;
+                if (!(s->dsize_mode == DETECTDSIZE_NE)) {
```

Comment:
could we support the NE operator here?

Think it could be cleaner to move the check into a small inline util func.

Reply:
Something along the lines of
`static inline bool SigDsizeModeNotEqual(const Signature *s) {
    return s->dsize_mode == DETECTDSIZE_NE;
}`
in detect-dsize.h perhaps?

Reply:
I was thinking about a bit wider scope of the inline func:
```c
static inline bool SigDsizePrefilter(...)
{
        if (unlikely(sflags & SIG_FLAG_DSIZE)) {
            if (likely(p->payload_len < s->dsize_low || p->payload_len > s->dsize_high)) {
                SCLogDebug("kicked out as p->payload_len %u, dsize low %u, hi %u",
                        p->payload_len, s->dsize_low, s->dsize_high);
                return false
            }
        }
        return true;
}
```
But then with the `NE` support added in as well.

Reply:
That makes way more sense. :)